src/search.rs

```diff
@@ -30,6 +30,23 @@ pub(crate) fn justfile(directory: &Path) -> Result<PathBuf, SearchError> {
   }
 }
 
+pub(crate) fn project_root(start: &Path) -> PathBuf {
+  for a in start.ancestors() {
+    let mut buf = a.to_path_buf();
+
+    for s in &[".git", "Cargo.toml"] {
+      buf.push(s);
+      if buf.exists() {
```

Comment:
I tend to avoid `Path::exists`, because it will mask permission errors, i.e. a permission error will yield `false`. It's verbose, but instead I like to do `Path::metadata` and look for `io::ErrorKind::NotFound`. (We can use `io::Result<PathBuf>`for the type of this function.)

Using a mutable path buf with pushing and popping saves a couple allocations per ancestor, but I think something that uses `.join` on each ancestor would be cleaner:


```
pub(crate) fn project_root(start: &Path) -> io::Result<&Path> {
  for directory in start.ancestors() {
    for name in &[".git", "Cargo.toml"] {
      match directory.join(name).metadata() {
        Ok(_) => {
          return Ok(directory);
        }
        Err(err) => {
          if err.kind != io::ErrorKind::NotFound {
            return Err(err);
          }
        }
      }
    }
  }

  start
}
```

Reply:
Should be resolved now.